File: EY.KnightsOfTheDebuggingTable.ProjectManagementPortal.Server/DTOs/ProjectDTO.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace EY.KnightsOfTheDebuggingTable.ProjectManagementPortal.Server.DTOs
{
    public class ProjectDto
    {
        public Guid? Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public Guid? CreatedBy { get; set; }
        public Guid TemplateId { get; set; }
        public Guid OwnerId { get; set; }
        public StageDto CurrentStage { get; set; }

        public IEnumerable<TaskDto> Tasks { get; set; }
        public IEnumerable<ResourceDto> Resources { get; set; } 
        public IEnumerable<StakeholderDto> Stakeholders { get; set; }
    }
}

File: ey.knightsofthedebuggingtable.projectmanagementportal.client/src/models/ProjectDto.ts
import type { StageDto } from "./StageDto";
import type { ResourceDto } from "./ResourceDto";
import type { StakeholderDto } from "./StakeholderDto";
import type { TaskDto } from "./TaskDto";

export type ProjectDto = {
  id?: string;
  name: string;
  description?: string;
  startDate: Date;
  endDate?: Date;
  createdBy?: string;
  templateId: string;
  ownerId: string;

  currentStage?: StageDto;
  stakeholders?: StakeholderDto[];
  tasks?: TaskDto[];
  resources?: ResourceDto[];
};
